usr.sbin/kldxref/ef.h

```diff
@@ -73,28 +82,48 @@ typedef int (elf_reloc_t)(struct elf_file *ef, const void *reldata,
     Elf_Type reltype, GElf_Addr relbase, GElf_Addr dataoff, size_t len,
     void *dest);
 
+typedef int (elf_capreloc_t)(struct elf_file *ef, const Gcapreloc *cr,
+    GElf_Addr relbase, GElf_Addr dataoff, size_t len, void *dest);
+
 struct elf_reloc_data {
 	unsigned char class;
 	unsigned char data;
 	GElf_Half machine;
 	elf_reloc_t *reloc;
 };
 
+struct elf_capreloc_data {
+	unsigned char class;
+	unsigned char data;
+	GElf_Half machine;
+	elf_capreloc_t *capreloc;
+};
+
 #define	ELF_RELOC(_class, _data, _machine, _reloc)			\
 	static struct elf_reloc_data __CONCAT(elf_reloc_data_, __LINE__) = { \
 	    .class = (_class),						\
 	    .data = (_data),						\
 	    .machine = (_machine),					\
-	    .reloc = (_reloc)						\
+	    .reloc = (_reloc),						\
```

Comment:
Slightly gratuitous to add the comma downstream (did you originally try and add a bool to elf_reloc_data saying it was a capreloc?)

Reply:
Oh, I can clean this up.  Originally I added a .capreloc member here rather than having a separate linker set, but the separate linker set is a cleaner downstream diff that can go away more easily when caprelocs go away.